### PR TITLE
Better composer install in the docker

### PR DESCRIPTION
**Pull request by tobybatch**
## Description
This switches the docker build to use the approved programmatic method for installing composer.

https://getcomposer.org/doc/faqs/how-to-install-composer-programmatically.md

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -17,24 +14,36 @@ RUN php -r "readfile('https://getcomposer.org/installer');" | php -- --install-d
...
+    apt remove -y wget && \
     docker-php-ext-install \
         gd \
         intl \
         ldap \
         pdo_mysql \
         zip && \
-    apt-get -y autoremove && \
-    apt-get clean && \
+    apt remove -y wget && \
```
Comment (kevinpapst) [inline Dockerfile]: this line looks like a duplicate

File: Dockerfile [hunk 2]
```diff
@@ -17,24 +14,36 @@ RUN php -r "readfile('https://getcomposer.org/installer');" | php -- --install-d
...
+        wget \
         zip \
         && \
+    EXPECTED_SIGNATURE="$(wget -q -O - https://composer.github.io/installer.sig)" && \
+    php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');" && \
+    ACTUAL_SIGNATURE="$(php -r "echo hash_file('sha384', 'composer-setup.php');")" && \
+    if [ "$EXPECTED_SIGNATURE" != "$ACTUAL_SIGNATURE" ]; then >&2 echo 'ERROR: Invalid installer signature'; rm composer-setup.php; exit 1; fi && \
+    php composer-setup.php --quiet && \
+    rm composer-setup.php && \
+    mv /var/www/html/composer.phar /usr/bin/composer && \
```
Comment (kevinpapst) [inline Dockerfile]: `chmod +x` is not required?

Comment (tobybatch) [reply Dockerfile]: Installer does this now so not needed.

**Comment by kevinpapst** [comment]
Perfect, thanks!

**Pull request merged**